src/ndarray/ndarray.cc

```diff
@@ -1128,7 +1128,7 @@ void CopyFromToImpl(const NDArray& from, const NDArray& to,
 }
 
 void CopyFromTo(const NDArray& from, const NDArray& to, int priority) {
-  if (from.var() == to.var()) {
+  if (from.var() == to.var() && from.byte_offset() == to.byte_offset()) {
```

Comment:
should we also check for size? @reminisce

Reply:
@piiswrong There is a shape equality check after this. For `CopyFromTo`, I think the added condition is sufficient. Agree we should also check sizes if it is for a function of checking if two ndarrays are the same one.